decimal.go

```diff
@@ -800,6 +812,30 @@ func (nd *NullDecimal) Scan(value interface{}) error {
 	return nd.Decimal.Scan(value)
 }
 
+var jsonNull = []byte("null")
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+func (nd *NullDecimal) UnmarshalText(b []byte) error {
+	if bytes.Compare(b, jsonNull) == 0 {
+		nd.Valid = false
+		return nil
+	}
+	if err := nd.Decimal.UnmarshalText(b); err != nil {
+		nd.Valid = false
+		return err
+	}
+	nd.Valid = true
+	return nil
+}
+
+// MarshalText implements the encoding.TextMarshaler interface.
+func (nd *NullDecimal) MarshalText() ([]byte, error) {
+	if nd.Valid {
+		return nd.Decimal.MarshalText()
+	}
+	return []byte("null"), nil
```

Comment:
Ugh... It's been a while since I've used github's review system. This comment keeps disappearing...

@mjibson It's unclear to me what the correct encoding of `null` is for `NullDecimal`. `null` will work for JSON, but it may not for other text encodings. It's easy enough to support multiple encodings of `null` in `UnmarshalText()`, but `encoding.TextMarshaler` doesn't provide an interface for configuring encodings for `MarshalText()`

Reply:
I agree this is tricky. I think for now we should remove these methods from the NullDecimal type to force the user to decide how to handle this situation, or at least until this project gets more information about what users expect.

Reply:
What do you think about implementing the following possible combinations of encoding interfaces on `NullDecimal`:
1. `json.Unmarshaler` and `json.Marshaler`
    * I'm a fan of this approach since it implements complementary interfaces and allows users to use `NullDecimal` with JSON
1. `encoding.TextUnmarshaler` and `json.Marshaler`
    * Supports generic text unmarshaling, but only supports marshaling to JSON

Reply:
How do we distinguish between a JSON null (not present, which shouldn't change `d` at all) and a SQL null, which should set `d` to the not valid state? That's why I think we shouldn't implement anything on NullDecimal.

Reply:
I was under the impression that if the value of `NullDecimal.Valid` is `false` then the proper marshaled JSON value would be `null`. This works out if we implement the `json.Marshaler` interface (e.g. the `MarshalJSON()` method) instead of the `encode.TextMarshaler` interface (e.g. the `MarshalText()` method)